### PR TITLE
Introduce NO_AUTO_CONSUME flag

### LINKED ISSUE #53997: Potions from Magiclysm are considered valid food for auto-eat behaviour
### Describe the bug

If you have potions of mana, cat's grace, ogre's strength and presumably others available nearby and have Auto-Eat zone for your character to feed, those potions are considered valid food and character would eat them up.

I don't know what/how priorities are given to different foods for purposes of auto-eat zones. Distance?

 I've had this set up (approximately):

100 smoked meats located 5 tiles away 
10 potions of mana located 6 tiles away
100 dehydrated meats located 6 tiles away

I've stayed in the location for a couple weeks, and had Auto-Eat zone. Over that couple weeks, character ate all smoked meats and then started eating potions with no desire to eat dehydrated meats instead. 

### Steps to reproduce

1. Make a valid auto-eat zone
2. Put potions nearby
3. Observe character eat potions

### Expected behavior

My expectations would be that Potions are not eaten automatically. Otherwise, see additional context for some more thoughts. 

### Screenshots

_No response_

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: 1.4; Distributor ID: Arch; Description: Arch Linux; Release: rolling; Codename: n/a; 
- Game Version: 75a01ca [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Aftershock [aftershock],
    Magiclysm [magiclysm],
    Blaze Industries [blazeindustries],
    DinoMod [DinoMod],
    Disable NPC Needs [no_npc_food],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]

### Additional context

I'm not sure what and how is it currently implemented, so I'll describe what I'd expect from an auto-eat zone.

Let's say you have a 11x11 space covered by auto-eat zone. On edge of that space, there is smoked meat which has 2 days till it rots. Closer to player, there is smoked meat which has 2 weeks to rot. Even closer to player, there are 2 piles dehydrated meat which has 3 seasons to rot, one being one tile closer. On same distance, there is something sealed which doesn't have an expiry date. 

I'd expect character to eat furtherest (and closest to rot) meat first, then move on to smoked meat which is still fastest to rot compared to other options. After that, dehydrated meat comes and closest pile is eaten first. Last of all is sealed food. 

Since most potions don't have expiry date on them, I'd expect them to be eaten only as the last resort if eating them can't be avoided in implementation of auto-eat zones.

### PR DESCRIPTION
**Pull request by rhmkrutov**
#### Summary
Mods "NO_AUTO_CONSUME internal json flag to avoid automatically eating food with secondary effects"

#### Purpose of change
fix #53997

#### Describe the solution
Added NO_AUTO_CONSUME flag to be defined for foods which primary goal is not supposed to be quenching or filling characters

#### Describe alternatives you've considered
Lowering calories, lowering fun value of the food.

#### Testing
Spawned potions, added auto-eat zone, gave myself hunger, read a book, potions were not eaten.

Spawned  a dehydrated meat, read a book in same spot, meat was eaten.

#### Additional context
May be should be named differently (HAS_SECONDARY_EFFECTS or something in lines)

### CODE REVIEW

File: src/activity_item_handling.cpp [hunk 1]
```diff
@@ -95,6 +95,8 @@ static const activity_id ACT_VEHICLE_REPAIR( "ACT_VEHICLE_REPAIR" );
 
 static const efftype_id effect_incorporeal( "incorporeal" );
 
+static const flag_id flag_NO_AUTO_EAT( "NO_AUTO_EAT" );
```
Comment (ZhilkinSerg) [inline src/activity_item_handling.cpp:98]:
```suggestion
static const flag_id json_flag_NO_AUTO_EAT( "NO_AUTO_EAT" );
```

File: src/activity_item_handling.cpp [hunk 2]
```diff
@@ -3173,6 +3175,10 @@ int get_auto_consume_moves( Character &you, const bool food )
         }
         for( item *it : items_here ) {
             item &comest = you.get_consumable_from( *it );
+            if (comest.has_flag( flag_NO_AUTO_EAT )) {
```
Comment (ZhilkinSerg) [inline src/activity_item_handling.cpp:3178]:
```suggestion
            if( comest.has_flag( json_flag_NO_AUTO_EAT ) ) {
```

File: src/activity_item_handling.cpp [hunk 3]
```diff
@@ -3173,6 +3175,10 @@ int get_auto_consume_moves( Character &you, const bool food )
...
+              // ignored due to auto-eat flag
```
Comment (ZhilkinSerg) [inline src/activity_item_handling.cpp:3179]:
```suggestion
              // ignored due to NO_AUTO_EAT flag
```

File: src/activity_item_handling.cpp [hunk 4]
```diff
@@ -3173,6 +3175,10 @@ int get_auto_consume_moves( Character &you, const bool food )
         }
         for( item *it : items_here ) {
             item &comest = you.get_consumable_from( *it );
+            if( comest.has_flag( json_flag_NO_AUTO_EAT ) ) {
+              // ignored due to NO_AUTO_EAT flag
+              continue;
```
Comment (ZhilkinSerg) [inline src/activity_item_handling.cpp:3180]:
```suggestion
                // ignored due to NO_AUTO_EAT flag
                continue;
```

File: data/json/flags.json [hunk 5]
```diff
@@ -568,6 +568,11 @@
     "type": "json_flag",
     "info": "This item used to be legal tender before the Cataclysm."
   },
+  {
+    "id": "NO_AUTO_EAT",
+    "type": "json_flag",
+    "//": "consumables with this flag should get ignored by auto-eat/auto-drink zones"
```
Comment (ZhilkinSerg) [inline data/json/flags.json:574]:
```suggestion
    "type": "json_flag"
```

**Comment by ZhilkinSerg** [comment]
> May be should be named differently (HAS_SECONDARY_EFFECTS or something in lines)

I would name it `NO_AUTO_CONSUME`

**Comment by NetSysFire** [comment]
I hope this is not out of scope since this is not related to Magiclysm, but apparently chocolate-covered coffee beans might also benefit from this flag: https://www.reddit.com/r/cataclysmdda/comments/rh0s0c/how_i_discovered_that_sweet_treats_are_worse_than/

**Comment by mkrutov** [comment]
@NetSysFire , would gladly add everything which is suggested there. I did go through items and didn't find anything which might benefit from it. If anything else is, please do share and I'd update the jsons.

**Comment by mkrutov** [comment]
@NetSysFire , added this to caffeine/stimulants &marloss stuff. didn't add it to Alcohol since there is specific check for alcohol addiction for auto consumption of alcohol beverages; and I didn't ever playtest alcoholics to see how it works as is.

**Pull request merged**